register/models.py

```diff
@@ -122,8 +139,14 @@ def send_reimbursement(self, request):
         self.save()
 
     def confirm(self, cancellation_url):
+        if self.status in [APP_ACCEPTED,APP_PENDING,APP_REJECTED]:
+            raise Http404('Unfortunately this invite doesn\'t even exist')
```

Comment:
In those three cases the hacker hasn't been invited (even if he's been accepted, since there's a time gap between acceptance and actual invitation). So that's why the _invite doesn't exist_. It took me a bit to work this out too so I'd welcome a better phrasing of the error.